cmd/darknode/darknode.go

```diff
@@ -190,25 +190,23 @@ func main() {
 		// Bootstrap into the network
 		fmtStr := "bootstrapping\n"
 		for _, bootstrapMulti := range config.BootstrapMultiAddresses {
-			if bootstrapMulti.Address() == multiAddr.Address() {
-				continue
-			}
-			multi, err := store.SwarmMultiAddressStore().MultiAddress(bootstrapMulti.Address())
-			if err != nil && err != swarm.ErrMultiAddressNotFound {
-				logger.Network(logger.LevelError, fmt.Sprintf("cannot get bootstrap multi-address from store: %v", err))
-				continue
-			}
-			if err == nil {
-				bootstrapMulti.Nonce = multi.Nonce
-				bootstrapMulti.Signature = multi.Signature
-			}
-			if err := store.SwarmMultiAddressStore().InsertMultiAddress(bootstrapMulti); err != nil {
-				logger.Network(logger.LevelError, fmt.Sprintf("cannot store bootstrap multiaddress in store: %v", err))
-			}
 			fmtStr += "  " + bootstrapMulti.String() + "\n"
+			_, err := store.SwarmMultiAddressStore().MultiAddress(bootstrapMulti.Address())
+			if err != nil {
+				if err == swarm.ErrMultiAddressNotFound {
+					if err := store.SwarmMultiAddressStore().InsertMultiAddress(multiAddr); err != nil {
+						logger.Network(logger.LevelError, fmt.Sprintf("cannot store bootstrap multiaddress in store: %v", err))
+					}
+				} else {
+					logger.Network(logger.LevelError, fmt.Sprintf("cannot get bootstrap multi-address from store: %v", err))
+					continue
+				}
+			}
```

Comment:
Same question here: what if the bootstrap multi-address has changed and we have updated the new change in it's multi-address in the config. Wouldn't this prevent the store from updating it just because the address was already present?

Reply:
Pushed code to check if the IP address of the bootstrap node has been changed.